ipa-core/src/protocol/context/mod.rs

```diff
@@ -917,6 +956,145 @@ mod tests {
         });
     }
 
+    #[test]
+    fn reshard_try_stream_basic() {
+        run(|| async move {
+            const SHARDS: u32 = 5;
+            let input: Vec<_> = (0..SHARDS).map(BA8::truncate_from).collect();
+            let world: TestWorld<WithShards<5>> =
+                TestWorld::with_shards(TestWorldConfig::default());
+            let r = world
+                .semi_honest(input.clone().into_iter(), |ctx, shard_input| async move {
+                    reshard_try_stream(ctx, stream::iter(shard_input).map(Ok), |_, record_id, _| {
+                        ShardIndex::from(u32::from(record_id) % SHARDS)
+                    })
+                    .await
+                    .unwrap()
+                })
+                .await
+                .into_iter()
+                .flat_map(|v| v.reconstruct())
+                .collect::<Vec<_>>();
+
+            assert_eq!(input, r);
+        });
+    }
+
+    #[test]
+    fn reshard_try_stream_less_items_than_expected() {
```

Comment:
what happens in the inverse? e.g., stream has more items than expected?

Reply:
nothing great, but definitely deserves a test to demonstrate that